### ISSUE TITLE
Interaction between process_children_only & autosnap/monitor

### ISSUE THREAD

**Issue by LordAro** [jimsalterjrs/sanoid#144]
I think this is an issue but I'm not *entirely* sure, so I'm interested in your opinion here :)

Currently, monitor_snapshots and take_snapshots both check that if autosnap/monitor is set to "no", or if process_children_only is set to "yes", then don't bother processing further.

However, if I set autosnap/monitor to "no", **and** process_children_only to "yes", then I do want this root folder to be processed, but not its children

Basically, should autosnap/monitor apply before or after process_children_only? I'm thinking the latter.

**Comment by jimsalterjrs** [comment]
I'm not 100% certain I'm following this.

If you set process_children_only to true, then you aren't doing anything with the dataset named in the module, period, no ifs ands or buts about it.

> I do want this root folder to be processed, but not its children

Then just don't set `recursive` on, and the children won't be touched. Recursive is off by default.

**Comment by phreaker0** [comment]
I think this is resolved? If not feel free to reopen.
